securityscanutils/securityscan.go

```diff
@@ -0,0 +1,284 @@
+
+	client, err := githubutils.GetClient(ctx)
+	if err != nil {
+		return eris.Wrap(err, "error initializing github client")
+	}
+	for _, repo := range s.Repos {
+		opts := repo.Opts
+		allReleases, err := githubutils.GetAllRepoReleases(ctx, client, repo.Owner, repo.Repo)
+		if err != nil {
+			return eris.Wrapf(err, "unable to fetch all github releases for github.com/%s/%s", repo.Owner, repo.Repo)
+		}
+		githubutils.SortReleasesBySemver(allReleases)
+		// Filter releases by version constraint provided
+		filteredReleases := githubutils.FilterReleases(allReleases, opts.VersionConstraint)
+		markdownTplFile, err := GetTemplateFile(MarkdownTrivyTemplate)
+		if err != nil {
+			return eris.Wrap(err, "error creating temporary markdown template file to pass to trivy")
+		}
+		sarifTplFile, err := GetTemplateFile(SarifTrivyTemplate)
+		if err != nil {
+			return eris.Wrap(err, "error creating temporary markdown template file to pass to trivy")
+		}
+		for _, release := range filteredReleases {
+			// We can swallow the error here, any releases with improper tag names
+			// will not be included in the filtered list
+			ver, _ := semver.NewVersion(release.GetTagName())
+			err = repo.RunMarkdownScan(ver, markdownTplFile)
+			if err != nil {
+				return eris.Wrapf(err, "error generating markdown file from security scan for version %s", release.GetTagName())
+			}
+			err = repo.RunGithubSarifScan(ver, sarifTplFile)
+			if err != nil {
+				return eris.Wrapf(err, "error generating github sarif file from security scan for version %s", release.GetTagName())
+			}
+		}
+
+	}
+	return nil
+}
+
+func (r *SecurityScanRepo) RunMarkdownScan(versionToScan *semver.Version, markdownTplFile string) error {
+	images, err := r.GetImagesToScan(versionToScan)
+	if err != nil {
+		return err
+	}
+	version := versionToScan.String()
+	outputDir := path.Join(r.Opts.OutputDir, r.Repo, "markdown_results", version)
+	err = osutils.CreateDirIfNotExists(outputDir)
+	if err != nil {
+		return err
+	}
+	for _, image := range images {
+		imageWithRepo := fmt.Sprintf("%s/%s:%s", r.Opts.ImageRepo, image, version)
+		fileName := fmt.Sprintf("%s_cve_report.docgen", image)
+		output := path.Join(outputDir, fileName)
+		_, err = RunTrivyScan(imageWithRepo, version, markdownTplFile, output)
+		if err != nil {
+			return eris.Wrapf(err, "error running image scan on image %s", imageWithRepo)
+		}
+	}
+	return nil
+}
+
+func (r *SecurityScanRepo) RunGithubSarifScan(versionToScan *semver.Version, sarifTplFile string) error {
+	images, err := r.GetImagesToScan(versionToScan)
+	if err != nil {
+		return err
+	}
+	version := versionToScan.String()
+	outputDir := path.Join(r.Opts.OutputDir, r.Repo, "sarif_results", version)
+	err = osutils.CreateDirIfNotExists(outputDir)
+	if err != nil {
+		return err
+	}
+	for _, image := range images {
+		imageWithRepo := fmt.Sprintf("%s/%s:%s", r.Opts.ImageRepo, image, version)
+		fileName := fmt.Sprintf("%s_cve_report.sarif", image)
```

Comment:
i don't see any harm in keeping it around, especially if someone were to generate this locally. I've already shown that this will be part of the output directory [here](https://github.com/solo-io/go-utils/blob/108a37f922d9598980952134bf659310d78ff694/securityscanutils/securityscan.go#L44). I estimate all the sarif files will take up ~2 MB of memory after even a 100 scans so it shouldn't be bad as far as resource usage.